### ISSUE TITLE
[MuonGun] uninitialized variables (Trac #1665)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1657]
```text
In file included from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h
    :13:0,
from /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:
    /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-
    services/surfaces/detail/CylinderBase.h: In member function ‘void
    I3MuonGun::Cylinder::serialize(boost::archive::xml_oarchive &, unsigned
        int)’:
/home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-services/surfaces/detail/CylinderBase.h:175:29:
    warning: ‘length’ may be used uninitialized in this function [-Wmaybe-
    uninitialized]
void SetLength(double v) { length_ = v; }
^~~~~~~
/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:18:
    note: ‘length’ was declared here
double radius, length;
^~~~~~
In file included from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h
    :13:0,
from /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:
    /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-services/surfaces/detail/CylinderBase.h:178:29:
    warning: ‘radius’ may be used uninitialized in this function [-Wmaybe-
    uninitialized]
void SetRadius(double v) { radius_ = v; }
^~~~~~~
/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:10:
    note: ‘radius’ was declared here
double radius, length;
^~~~~~
In file included from
    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/serializati
    on.h:39:0,
from /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-
    services/surfaces/Surface.h:13,
from /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-
    services/surfaces/SamplingSurface.h:12,
from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/public/MuonGun/SamplingSur
    face.h:12,
from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h
    :12,
from /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:
    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary
    _archive.hpp: In member function ‘void I3MuonGun::Cylinder::serialize(
        boost::archive::portable_binary_oarchive &, unsigned int)’:
/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary_archive.hpp:139:11:
    warning: ‘length’ may be used uninitialized in this function [-Wmaybe-
    uninitialized]
T st = t;
^
/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:18:
    note: ‘length’ was declared here
double radius, length;
^~~~~~
In file included from
    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/serializati
    on.h:39:0,
from /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-
    services/surfaces/Surface.h:13,
from /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-
    services/surfaces/SamplingSurface.h:12,
from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/public/MuonGun/SamplingSur
    face.h:12,
from
    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h
    :12,
from /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:
    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary_archive.hpp:139:11:
    warning: ‘radius’ may be used uninitialized in this function [-Wmaybe-
    uninitialized]
T st = t;
^
/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:10:
    note: ‘radius’ was declared here
double radius, length;
^~~~~~
```

Migrated from https://code.icecube.wisc.edu/ticket/1665
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:35", 
    "description": "{{{\nIn file included from\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h\n    :13:0,\nfrom /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:\n    /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-\n    services/surfaces/detail/CylinderBase.h: In member function \u2018void\n    I3MuonGun::Cylinder::serialize(boost::archive::xml_oarchive &, unsigned\n        int)\u2019:\n/home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-services/surfaces/detail/CylinderBase.h:175:29:\n    warning: \u2018length\u2019 may be used uninitialized in this function [-Wmaybe-\n    uninitialized]\nvoid SetLength(double v) { length_ = v; }\n^~~~~~~\n/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:18:\n    note: \u2018length\u2019 was declared here\ndouble radius, length;\n^~~~~~\nIn file included from\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h\n    :13:0,\nfrom /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:\n    /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-services/surfaces/detail/CylinderBase.h:178:29:\n    warning: \u2018radius\u2019 may be used uninitialized in this function [-Wmaybe-\n    uninitialized]\nvoid SetRadius(double v) { radius_ = v; }\n^~~~~~~\n/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:10:\n    note: \u2018radius\u2019 was declared here\ndouble radius, length;\n^~~~~~\nIn file included from\n    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/serializati\n    on.h:39:0,\nfrom /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-\n    services/surfaces/Surface.h:13,\nfrom /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-\n    services/surfaces/SamplingSurface.h:12,\nfrom\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/public/MuonGun/SamplingSur\n    face.h:12,\nfrom\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h\n    :12,\nfrom /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:\n    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary\n    _archive.hpp: In member function \u2018void I3MuonGun::Cylinder::serialize(\n        boost::archive::portable_binary_oarchive &, unsigned int)\u2019:\n/home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary_archive.hpp:139:11:\n    warning: \u2018length\u2019 may be used uninitialized in this function [-Wmaybe-\n    uninitialized]\nT st = t;\n^\n/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:18:\n    note: \u2018length\u2019 was declared here\ndouble radius, length;\n^~~~~~\nIn file included from\n    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/serializati\n    on.h:39:0,\nfrom /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-\n    services/surfaces/Surface.h:13,\nfrom /home/dschultz/Documents/combo/trunk/src/phys-services/public/phys-\n    services/surfaces/SamplingSurface.h:12,\nfrom\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/public/MuonGun/SamplingSur\n    face.h:12,\nfrom\n    /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.h\n    :12,\nfrom /home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:9:\n    /home/dschultz/Documents/combo/trunk/src/icetray/public/icetray/portable_binary_archive.hpp:139:11:\n    warning: \u2018radius\u2019 may be used uninitialized in this function [-Wmaybe-\n    uninitialized]\nT st = t;\n^\n/home/dschultz/Documents/combo/trunk/src/MuonGun/private/MuonGun/Cylinder.cxx:53:10:\n    note: \u2018radius\u2019 was declared here\ndouble radius, length;\n^~~~~~\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067215093672", 
    "component": "combo simulation", 
    "summary": "[MuonGun] uninitialized variables", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-04-26T20:53:34", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160729T06:41:02: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "new" to "closed"
* jvansanten commented:

> In changeset 148884:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="148884"
> Fix #1657
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:35: 
* olivas commented:

> Milestone IceSim 5.1 deleted


* olivas changed milestone from "IceSim 5.1" to ""
